test/test_amnesty.py

```diff
@@ -0,0 +1,297 @@
+"""
+Tests of the lint-amnesty command.
+"""
+
+from collections import defaultdict
+import itertools
+import textwrap
+import unittest
+
+from edx_lint.cmd.amnesty import PylintError, fix_pylint, PYLINT_EXCEPTION_REGEX, parse_pylint_output
+
+class AmnestyTest(unittest.TestCase):
+    """
+    Tests of lint-amnesty.
+    """
+
+    def assert_amnesty(self, input_code, errors, expected):
+        """
+        Assert that fix_pylint produces ``expected`` when fed ``input_code`` and the
+        list of errors ``errors``.
+
+        Arguments:
+            input_code: A string of python code. Will be textwrap.dedented.
+            errors: A list of PylintErrors
+            expected: A string of python code. Will be textwrap.dedented.
+        """
+        input_code = textwrap.dedent(input_code)
+        expected = textwrap.dedent(expected)
+
+        errors_by_line = defaultdict(list)
+        for error in errors:
+            errors_by_line[error.linenum].append(error)
+
+        output_lines = itertools.chain.from_iterable(
+            fix_pylint(line, errors_by_line[lineno])
+            for lineno, line
+            in enumerate(input_code.split('\n'), start=1)
+        )
+
+        self.assertEqual(expected.split('\n'), list(output_lines))
+
+    def assert_pylint_exception_match(self, expected, line):
+        """
+        Assert that PYLINT_EXCEPTION_REGEX mateches ``expected`` in ``line``.
+        """
+        self.assertEqual(
+            expected,
+            PYLINT_EXCEPTION_REGEX.search(line).group(0)
+        )
+
+    def test_pylint_exception_re(self):
+        self.assert_pylint_exception_match(
+            "  # pylint: disable=some-exception, other-exception",
+            "some line  # pylint: disable=some-exception, other-exception",
+        )
+        self.assert_pylint_exception_match(
+            "  # pylint: disable=some-exception",
+            "some line  # pylint: disable=some-exception",
+        )
+        self.assert_pylint_exception_match(
+            "  # pylint: disable=some-exception",
+            "some line  # pylint: disable=some-exception#noqa",
+        )
+        self.assert_pylint_exception_match(
+            "  # pylint: disable=some-exception",
+            "some line  # pylint: disable=some-exception  #noqa",
+        )
+        self.assert_pylint_exception_match(
+            "# pylint: disable=some-exception",
+            "some line# pylint: disable=some-exception#noqa",
+        )
+
+    def test_parse_pylint_output(self):
+        pylint_output = textwrap.dedent("""\
+            ************* Module cms.urls
+            cms/urls.py:1: [C0111(missing-docstring), ] Missing module docstring
+            ************* Module cms.envs.test
+            cms/envs/test.py:81: [C0301(line-too-long), ] Line too long (144/120)
+        """)
```

Comment:
Please verify a pylint violation whose parseable output spans multiple lines. 
E.g. if I take out the spaces around the equals on line 74 above, I get the following output.
```
************* Module test.test_amnesty
test/test_amnesty.py:78: [C0326(bad-whitespace), ] Exactly one space required around assignment
        pylint_output=textwrap.dedent("""\
                     ^
```

Also we should verify parsing with multiple violations per file.